src/services/formatting/smartIndenter.ts

```diff
@@ -464,9 +471,25 @@ namespace ts.formatting {
                 case SyntaxKind.GetAccessor:
                 case SyntaxKind.SetAccessor:
                     return child !== SyntaxKind.Block;
+                case SyntaxKind.ExportDeclaration:
+                    return child !== SyntaxKind.NamedExports;
                 default:
                     return false;
             }
         }
+
+        /**
+         * Function returns true if a node should not get additional indentation in its parent node.
+         */
+        export function isIndentationPrevented(node: TextRangeWithKind) {
+            switch (node.kind) {
+                case SyntaxKind.NamedExports:
+                    return true;
+                case SyntaxKind.ImportClause:
+                    // NamedImports has its own braces as Block does
+                    return (<ImportClause>node).namedBindings.kind === SyntaxKind.NamedImports;

```

Comment:
@vladima Former `hasBraceShell` (and current `namedBindings` check) is needed to prevent unwanted excessive indentation. Consider this code:

```typescript
import
{
    x as xx
}
from "foo"
import
    * as Foo from "foo"
```

ImportDeclaration has ImportClause node and then ImportClause node can have two different kind of nodes: `NamedImports` and `NamespaceImport`.

`NamedImports`, as you noted, looks like a block, but `NamespaceImport` doesn't. If I decide to always indent `ImportClause` or always not to do it, formatting will look bad:

```typescript
import
    {
        x as xx
    } // when ImportClause is always indented
from "foo"

import
* as Foo from "foo" // when always not
```

Thus I added this check.

Anyway I agree that isCompletedNode should be updated. I'll push a change for it.